### DISCUSSION TITLE
TDataset Generic in ParallelDataManager

### DISCUSSION THREAD

**Discussion by fortmeier** [nerfstudio-project/nerfstudio#2767] [Q&A] [answered]
Hi everyone, I starting to get into nerfstudio and encountered something I do not fully understand.

What I am trying to achieve is using the ParallelDataManager, but with another InputDataset.

In the ParallelDataManager, there is this line 
https://github.com/nerfstudio-project/nerfstudio/blob/69ed6d40935ced2a295702423dac8c0020e3a7d1/nerfstudio/data/datamanagers/parallel_datamanager.py#L144

My understanding of generics is limited, but it looks like self.dataset_type is either set by the _dataset_type or is set to the default of TDataset (InputDataset). Other datamanagers behave differently, for example VanillaDataManager uses a method for it:

https://github.com/nerfstudio-project/nerfstudio/blob/69ed6d40935ced2a295702423dac8c0020e3a7d1/nerfstudio/data/datamanagers/base_datamanager.py#L429C1-L452C23

Is this intentional?

Thanks.

**Comment by hturki** [comment, marked as answer]
https://github.com/nerfstudio-project/nerfstudio/pull/2884
